### PR TITLE
Add module variables for NCEPLIBS 

### PR DESCRIPTION
**Pull request by kgerheiser**
Use setup_run_environment to search for libraries and set env variables for module generation. 

Other non-spack builds rely on these variables.

Libraries are installed with CMAKE_INSTALL_LIBDIR, which can be lib or lib64 depending on the machine, which makes it impossible to hardcode through modules.yaml.

I know this is a little hacky,  but the documentation says to use setup_run_environment to add module variables, and I don't know of any other way to query the lib directory.

### CODE REVIEW

**Comment by jbathegit** [comment]
For now this will work, but as an FYI the next version of NCEPLIBS-bufr (11.6.0) will no longer have _DA builds.  I'm hoping to release 11.6.0 within the next few weeks, and then from that point on there will only be the 3 variants _4, _8 and _d, without any additional _DA variants.  See https://github.com/NOAA-EMC/NCEPLIBS-bufr/pull/155

**Comment by kgerheiser** [comment]
Thanks @jbathegit, I have added a check for versions less than equal to 11.5.0

File: var/spack/repos/builtin/packages/bufr/package.py [hunk 1]
```diff
@@ -21,3 +21,37 @@ class Bufr(CMakePackage):
...
 
     version('11.5.0', sha256='d154839e29ef1fe82e58cf20232e9f8a4f0610f0e8b6a394b7ca052e58f97f43')
+
+    def setup_run_environment(self, env):
+
+        lib4 = find_libraries('libbufr_4', root=self.prefix,
+            shared=False, recursive=True)
+        lib8 = find_libraries('libbufr_8', root=self.prefix, 
+            shared=False, recursive=True)
+        libd = find_libraries('libbufr_8', root=self.prefix,
```
Comment (t-brown) [inline var/spack/repos/builtin/packages/bufr/package.py]: Is the library suffix `8` and not `d`?

**Review by edwardhartnett — DISMISSED** [review]

**Review by Hang-Lei-NOAA — DISMISSED** [review]

**Review by t-brown — DISMISSED** [review]

File: var/spack/repos/builtin/packages/bufr/package.py (hunk 1 above)
Comment (kgerheiser) [reply var/spack/repos/builtin/packages/bufr/package.py]: Yes, good catch.

File: var/spack/repos/builtin/packages/bufr/package.py [hunk 2]
```diff
@@ -21,3 +21,37 @@ class Bufr(CMakePackage):
...
     version('11.5.0', sha256='d154839e29ef1fe82e58cf20232e9f8a4f0610f0e8b6a394b7ca052e58f97f43')
+
+    def setup_run_environment(self, env):
+
+        lib4 = find_libraries('libbufr_4', root=self.prefix,
+                              shared=False, recursive=True)
+        lib8 = find_libraries('libbufr_8', root=self.prefix,
+                              shared=False, recursive=True)
+        libd = find_libraries('libbufr_8', root=self.prefix,
+                              shared=False, recursive=True)
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/bufr/package.py]: Is this a typo:
```suggestion
        libd = find_libraries('libbufr_d', root=self.prefix,
                              shared=False, recursive=True)
```
?

Comment (alalazo) [reply var/spack/repos/builtin/packages/bufr/package.py]: Can we factor together the operations for a single library? For instance:
```python
def _setup_libufr_environment(self, env, suffix):
    libname = 'libufr_{0}'.format(suffix)
    ...

def setup_run_environment(self, env):
    for suffix in ('4', '8', 'd'):
        self._setup_libufr_environment(env, suffix)
```

**Review by alalazo — CHANGES_REQUESTED** [review]
There's a lot of repetition when setting up environments for libraries having different suffixes. Can we factor the operations together to reduce duplication?

File: var/spack/repos/builtin/packages/bufr/package.py (hunk 2 above)
Comment (kgerheiser) [reply var/spack/repos/builtin/packages/bufr/package.py]: Yes

Comment (kgerheiser) [reply var/spack/repos/builtin/packages/bufr/package.py]: Done, how does it look now?

**Review by Hang-Lei-NOAA — APPROVED** [review]

**Review by alalazo — APPROVED** [review]

**Pull request merged**